pkg/controller/deployment/deployment_controller.go

```diff
@@ -754,6 +742,158 @@ func (dc *DeploymentController) getNewReplicaSet(deployment *extensions.Deployme
 	return createdRS, dc.updateDeploymentRevision(deployment, newRevision)
 }
 
+// getNewReplicaSetAndSyncHashKey returns a replica set that matches the intent of the given deployment.
+// Returns nil if the new replica set doesn't exist yet.
+// Note that the pod-template-hash will be added to adopted RSes and pods.
+func (dc *DeploymentController) getNewReplicaSetAndSyncHashKey(deployment *extensions.Deployment, c clientset.Interface) (*extensions.ReplicaSet, error) {
```

Comment:
Let's try and keep the helpers at a minimum. These two calls in here could be inlined in `getNewReplicaSet`. Merge godoc if you wish.